test/Microsoft.AspNetCore.Mvc.FunctionalTests/BasicTests.cs

```diff
@@ -616,6 +616,18 @@ public async Task BindPropertiesAttributeWithSupportsGet_BindsOnGet()
             Assert.Equal("OnGetTestName", content);
         }
 
+        [Fact]
+        public async Task CanRunMiddlewareAfterRouting()
```

Comment:
This will only run with old routing. Move to a base class that runs with both old and new routing.

I think the routing website and tests would be the best place.